### PR TITLE
[rps] Fixes to the endgame

### PR DESCRIPTION
**Pull request by geoknee**
- [x] Insufficient funds flow works
- [x] Resigning flow works
- [x] And other bug fixes. 
- [x] Diagram updated:
- [ ] update again to show resigning flow
user-workflows-new-api

### CODE REVIEW

File: packages/rps/src/containers/GameContainer.tsx [hunk 1]
```diff
@@ -65,64 +71,73 @@ function RenderGame(props: GameProps) {
...
       );
+    case 'Resigned':
+      const ourTurn =
+        localState.player === 'A'
+          ? Number(channelState.turnNum) % 2 !== 0
+          : Number(channelState.turnNum) % 2 === 0;
+      const iResigned =
+        (ourTurn && channelState.status === 'running') ||
+        (!ourTurn && channelState.status === 'closing') ||
+        (ourTurn && channelState.status === 'closed');
```
Comment (tomclose) [inline packages/rps/src/containers/GameContainer.tsx]: I feel like this should be stored in the state. It seems like too much logic to be in the view layer.

Comment (geoknee) [reply packages/rps/src/containers/GameContainer.tsx]: Addressed in 4d98af6

File: packages/rps/src/redux/login/saga.ts [hunk 2]
```diff
@@ -66,5 +67,6 @@ export default function* loginRootSaga() {
 }
 
 function getLibraryAddress() {
-  return process.env.RPS_CONTRACT_ADDRESS;
+  return '0x0'; // TODO fix this
+  // return process.env.RPS_CONTRACT_ADDRESS;
```
Comment (geoknee) [inline packages/rps/src/redux/login/saga.ts]: I couldn't get this to work :-(

Comment (tomclose) [reply packages/rps/src/redux/login/saga.ts]: I think you need this: https://github.com/statechannels/monorepo/blob/master/packages/wallet/config/env.js#L47-L53

**Pull request merged**